lib/core/engine/iteration.js

```diff
@@ -61,6 +61,57 @@ class Iteration {
     this.asyncScriptsByCategory = asyncScriptsByCategory;
   }
 
+  async runPrePostScript(script, driver, context, commands) {
+    let requiresPrivilege = false;
+
+    if (typeof script === 'object') {
+      const scriptObject = script;
+      if (
+        typeof scriptObject.function !== 'function' ||
+        typeof scriptObject.requires !== 'object'
+      ) {
+        log.error(
+          'New-style pre/post script detected ' +
+            'but with invalid construction. ' +
+            'It will not execute.'
+        );
+        script = null;
+      } else {
+        log.info('Working with a new-style script.');
+        script = scriptObject.function;
+        requiresPrivilege =
+          scriptObject.requires.privilege === true ? true : false;
+      }
+    }
+    if (
+      requiresPrivilege &&
+      (typeof driver.getContext !== 'function' ||
+        typeof driver.setContext !== 'function')
+    ) {
+      log.error(
+        'Only browsers with get/setContext functions ' +
+          'can run privileged JavaScript.'
+      );
+      script = null;
+    }
+
+    if (script !== null) {
```

Comment:
All these `!== null` aren't particularly idiomatic.  Just `if (script) { ... }`.  If people give wacky inputs, they can deal with wacky outputs.

Reply:
is 

`if (x === null) {`

idiomatic or should that be 

`if (x) {`

?

Reply:
Generally just `if (x) { }`.  Sometimes the difference matters, but generally it doesn't; anytime there's an object equality one has to think about what exactly is being guarded against.  Here, it's just "is this set", so no need for strict object equality.